sonarts-sq-plugin/sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/externalreport/TslintRulesTest.java

```diff
@@ -0,0 +1,366 @@
+  private static final Path README_PATH = Paths.get("..", "..", "README.md");
+  private static final Path SONAR_TS_RULES_PATH = Paths.get("..", "..", "sonarts-core", "src", "rules");
+  private static final Path JAVA_ANNOTATED_RULES_PATH = Paths.get("src", "main", "java", "org", "sonar", "plugin", "typescript", "rules");
+
+  private static Set<String> javaAnnotatedRuleTsLintKeys;
+  private static Map<String, Rule> ruleByTsLintKeyFromReadMe;
+  private static Map<String, Rule> ruleByTsLintKeyFromSonarTS;
+  private static Map<String, Rule> ruleByTsLintKeyFromTsLintJson;
+  private static Map<String, Rule> ruleByTsLintKeyFromTsLintSonarTsJson;
+  private static Set<String> bugTslintRules;
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    javaAnnotatedRuleTsLintKeys = new JavaAnnotatedRules(JAVA_ANNOTATED_RULES_PATH).ruleTsLintKeys;
+    ruleByTsLintKeyFromReadMe = new ReadMeRules(README_PATH).ruleByTsLintKey;
+    ruleByTsLintKeyFromSonarTS = new SonarTsRules(SONAR_TS_RULES_PATH).ruleByTsLintKey;
+    ruleByTsLintKeyFromTsLintJson = loadJson(TSLINT_RULES_JSON_PATH);
+    ruleByTsLintKeyFromTsLintSonarTsJson = loadJson(TSLINT_SONARTS_RULES_JSON_PATH);
+    bugTslintRules = new HashSet<>(Arrays.asList(
+      "await-promise",
+      "ban-comma-operator",
+      "ban",
+      "curly",
+      "forin",
+      "import-blacklist",
+      "label-position",
+      "no-arg",
+      "no-bitwise",
+      "no-conditional-assignment",
+      "no-console",
+      "no-construct",
+      "no-debugger",
+      "no-duplicate-super",
+      "no-duplicate-switch-case",
+      "no-duplicate-variable",
+      "no-dynamic-delete",
+      "no-empty",
+      "no-eval",
+      "no-floating-promises",
+      "no-for-in-array",
+      "no-implicit-dependencies",
+      "no-inferred-empty-object-type",
+      "no-invalid-template-strings",
+      "no-invalid-this",
+      "no-misused-new",
+      "no-null-keyword",
+      "no-object-literal-type-assertion",
+      "no-return-await",
+      "no-shadowed-variable",
+      "no-sparse-arrays",
+      "no-string-literal",
+      "no-string-throw",
+      "no-submodule-imports",
+      "no-switch-case-fall-through",
+      "no-this-assignment",
+      "no-unbound-method",
+      "no-unnecessary-class",
+      "no-unsafe-any",
+      "no-unsafe-finally",
+      "no-unused-expression",
+      "no-unused-variable",
+      "no-use-before-declare",
+      "no-var-keyword",
+      "no-void-expression",
+      "prefer-conditional-expression",
+      "prefer-object-spread",
+      "radix",
+      "restrict-plus-operands",
+      "strict-boolean-expressions",
+      "strict-type-predicates",
+      "switch-default",
+      "triple-equals",
+      "typeof-compare",
+      "use-default-type-parameter",
+      "use-isnan"));
+  }
+
+  @Test
+  public void all_java_annotated_rules_exist_in_tslint_rules_json() {
```

Comment:
I don't think this asserting is important

Reply:
I disagree, for example: if a user disable @Rule(key = "S1301") in SQ quality profile, I want to ensure that we will import an external issue, with a human readable name described in `rules.json`.